### PR TITLE
Update actions in workflows to latest versions

### PR DESCRIPTION
**Pull request by flodolo**

### CODE REVIEW

File: .github/workflows/py-lint.yml [hunk 1]
```diff
@@ -49,8 +49,8 @@ jobs:
   pyupgrade:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
         with:
           python-version: 3.9
```
Comment (flodolo) [inline .github/workflows/py-lint.yml:55]: Is there a reason to keep using 3.9, and not a more recent version (e.g. 3.12)? I noticed black reformats several files locally, but not in the workflow.

Comment (mathjazz) [reply .github/workflows/py-lint.yml:55]: No, we should upgrade. We might even _need_ to upgrade in order to land #3100.

**Review by mathjazz — APPROVED** [review]

File: .github/workflows/py-lint.yml (hunk 1 above)
Comment (flodolo) [reply .github/workflows/py-lint.yml:55]: I'll do it in a separate PR to see if the workflows actually fail as I would expect.

Comment (mathjazz) [reply .github/workflows/py-lint.yml:55]: Makes sense, we should do that also for Docker and Heroku in one go:
https://github.com/mozilla/pontoon/blob/main/docker/Dockerfile
https://github.com/mozilla/pontoon/blob/main/runtime.txt

**Pull request merged**